From 463945facb237b59482bff1b7a2d0a74d26d9a0a Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Tue, 24 Jan 2023 12:49:57 +0200
Subject: [PATCH] [SW-112019] greco: security: fix efuse bit masks

In Greco there are separate efuse bits for security enable(bit 15) and
image verification(bit 10)

Created different functions for security enabled and image verification

Change-Id: I9ebe0fd8ce5a1e9e6e17ab398264909daf98e043
Signed-off-by: Sailaja <sbandarupalli@habana.ai>
---

diff --git a/arch/greco/arc/security.c b/arch/greco/arc/security.c
index df67a07..0d3a9a5 100644
--- a/arch/greco/arc/security.c
+++ b/arch/greco/arc/security.c
@@ -15,6 +15,7 @@
 #include <efuse.h>
 
 #define EFUSE_SYSCFG_ENABLE_VALIDATION_MASK	BIT(10)
+#define EFUSE_SYSCFG_ENABLE_SECURITY_MASK	BIT(15)
 
 u32 sp_sram_readl(u32 offset)
 {
@@ -31,7 +32,19 @@
 	if (ret)
 		return false;
 
-	return (config_efuse & EFUSE_SYSCFG_ENABLE_VALIDATION_MASK);
+	return !!(config_efuse & EFUSE_SYSCFG_ENABLE_SECURITY_MASK);
+}
+
+bool preboot_img_validation_is_enabled(void)
+{
+	u32 config_efuse;
+	int ret;
+
+	ret = preboot_fuse_read(EFUSE_SYSCFG_OFFSET, &config_efuse);
+	if (ret)
+		return false;
+
+	return !!(config_efuse & EFUSE_SYSCFG_ENABLE_VALIDATION_MASK);
 }
 
 bool preboot_security_key_is_revoked(int key_idx)
